test/end2end_test.go

```diff
@@ -7668,3 +7668,82 @@ func (s) TestClientSettingsFloodCloseConn(t *testing.T) {
 	s.GracefulStop()
 	timer.Stop()
 }
+
+// TestDeadlineSetOnConnectionOnClientCredentialHandshake tests that there is a deadline
+// set on the net.Conn when a credential handshake happens in http2_client.
+func (s) TestDeadlineSetOnConnectionOnClientCredentialHandshake(t *testing.T) {
+	lis, err := net.Listen("tcp", "localhost:50053")
+	if err != nil {
+		t.Fatalf("Failed to listen: %v", err)
+	}
+	go func() {
+		lis.Accept()
+	}()
+	credsInvoked := testutils.NewChannel()
+	cvd := &credentialsVerifyDeadline{
+		credsInvoked: credsInvoked,
+	}
```

Comment:
Let's actually push the deadline itself onto the channel and move the verifying into the test.  It will make the creds implementation more useful for other potential things, and locates everything related to this test case inside this test case.

Reply:
Changed.